pkg/reconciler/pipeline/dag/dag_test.go

```diff
@@ -669,3 +669,59 @@ func TestBuild_ConditionsParamsFromTaskResults(t *testing.T) {
 	}
 	assertSameDAG(t, expectedDAG, g)
 }
+
+// This test make sure we detect cycle when it exists.
+// This means we "visit" "a" twice, from two different paths.
+// And one of the paths is creating a cycle with "e -> a".
+//   a
+//  / \
+// b   c
+//  \ /
+//   d
+//  / \
+// e   f
+//     |
+//     g
+func TestBuild_InvalidDAGWithCycle(t *testing.T) {
```

Comment:
I'm not 100% sure, but should we duplicate this test in `pkg/reconciler/pipeline/dag/dagv1beta1_test.go` as well?

Reply:
sure I added in `v1beta1` as well